root/static/scripts/work.js

```diff
@@ -121,36 +121,46 @@ class WorkAttribute {
     this.attributeValue = ko.observable(data.field.value.value);
     this.errors = ko.observableArray(subfieldErrors(data));
     this.parent = parent;
+    this.previousTypeID;
     this.typeHasFocus = ko.observable(false);
     this.typeID = ko.observable(data.field.type_id.value);
 
     this.allowedValues = ko.computed(() => {
       const typeID = this.typeID();
 
-      if (!typeID || this.allowsFreeText()) {
+      if (this.allowsFreeText(typeID)) {
         return [];
       }
       return this.parent.allowedValuesByTypeID[typeID] ?? [];
     });
 
+    this.typeID.subscribe((previousTypeID => {
+      this.previousTypeID = previousTypeID;
+    }), this, 'beforeChange');
```

Comment:
This causes an issue with our flow typing for subscribe - the typing needs to be updated to support these extra arguments, AFAICT. Not quite sure how to do so.